### PR TITLE
Support IAsyncEnumerable for Amazon.DynamoDBv2.DataModel.AsyncSearch

### PR DESCRIPTION
**Pull request by sudoudaisuke**
Support IAsyncEnumerable for Amazon.DynamoDBv2.DataModel.AsyncSearch.

## Description
Implement IAsyncEnumerable for AsyncSearch in netcoreapp3.1.

I created pull request #1482. However, this request was rejected due to breaking changes. Now SDK support netcoreapp3.1 target. This request does not add a new target and a new dependency package.

#1468

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: sdk/src/Services/DynamoDBv2/Custom/DataModel/_async/AsyncSearch.IAsyncEnumerable.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2012-2013 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (normj) [inline sdk/src/Services/DynamoDBv2/Custom/DataModel/_async/AsyncSearch.IAsyncEnumerable.cs:2]: Remove the year. Just say "Copyright Amazon.com, ..."

**Comment by normj** [comment]
I agree now that we have .NET Core 3.1 support in the SDK we can add this.

Can you try and add a test in [DataModelTests.cs](https://github.com/aws/aws-sdk-net/blob/master/sdk/test/NetStandard/IntegrationTests/IntegrationTests/DynamoDB/DataModelTests.cs). I know this test project is challenging to work with given the size of the solution. Let me know if it doesn't work for you.

File: sdk/src/Services/DynamoDBv2/Custom/DataModel/_async/AsyncSearch.IAsyncEnumerable.cs (hunk 1 above)
Comment (sudoudaisuke) [reply sdk/src/Services/DynamoDBv2/Custom/DataModel/_async/AsyncSearch.IAsyncEnumerable.cs]: Thank you. I fixed this.

**Comment by sudoudaisuke** [comment]
> I agree now that we have .NET Core 3.1 support in the SDK we can add this.
> 
> Can you try and add a test in [DataModelTests.cs](https://github.com/aws/aws-sdk-net/blob/master/sdk/test/NetStandard/IntegrationTests/IntegrationTests/DynamoDB/DataModelTests.cs). I know this test project is challenging to work with given the size of the solution. Let me know if it doesn't work for you.

Hello normj. Thank you for your review.
I created a test. But sorry, I don't know how to run this test. Could you give me the documentation to run this?. Thank you.

**Comment by sudoudaisuke** [comment]
AWS .NET SDK starts support with IAsyncEnumerable (Paginators API). This pull request was not use paginator, so I close this issue.
I'm looking foward to IDynamoDBContext (object persistence model) supporting paginator.

**Pull request closed without merging**